src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt

```diff
@@ -125,111 +201,79 @@ abstract class Display() {
 
     fun show() = apply {
-        setShouldRender(true)
+        this.visible = true
+        this.lines.forEach { it.visible = true }
     }
 
     fun hide() = apply {
-        setShouldRender(false)
-    }
-
-    fun getWidth(): Float = width
-
-    fun getHeight(): Float = height
-
-    fun getMinWidth(): Float = minWidth
-
-    fun setMinWidth(minWidth: Float) = apply {
-        this.minWidth = minWidth
+        this.visible = false
+        this.lines.forEach { it.visible = false }
     }
 
     /**
-     * Gets the type of register the display will render under.
-     *
-     * The returned value will be a DisplayHandler.RegisterType
-     *      renderOverlayEvent: render overlay
-     *      postGuiRenderEvent: post gui render
+     * This register allows you to call a method right before the Display is drawn.
      *
-     * @return the register type
+     * @param method The method to call.
      */
-    fun getRegisterType(): DisplayHandler.RegisterType = registerType
+
+    fun registerPreDraw(method: Any) = run {
+        onPreDraw = RegularTrigger(method, TriggerType.Other, getLoader())
+        onPreDraw
+    }
 
     /**
-     * Sets the type of register the display will render under.
+     * This register allows you to call a method right after the Display is drawn.
      *
-     * Possible input values are:.
-     *      renderOverlayEvent: render overlay
-     *      postGuiRenderEvent: post gui render
+     * @param method The method to call.
      */
-    fun setRegisterType(registerType: Any) = apply {
-        this.registerType = when (registerType) {
-            is String -> DisplayHandler.RegisterType.valueOf(registerType.uppercase().replace(" ", "_"))
-            is DisplayHandler.RegisterType -> registerType
-            else -> DisplayHandler.RegisterType.RENDER_OVERLAY
-        }
+
+    fun registerPostDraw(method: Any) = run {
+        onPostDraw = RegularTrigger(method, TriggerType.Other, getLoader())
+        onPostDraw
     }
 
-    fun render() {
-        if (!shouldRender)
-            return
+    internal fun draw() {
+
+        if (!visible) return
+
+        onPreDraw?.trigger(arrayOf(this))
 
         width = lines.maxOfOrNull { it.getTextWidth() }?.coerceAtLeast(minWidth) ?: minWidth
 
         var i = 0f
         lines.forEach {
-            it.draw(renderX, renderY + i, width, background, backgroundColor, textColor, align)
+            it.draw(x, y + i, width, backgroundType, backgroundColor, textColor, align)
 
             when (order) {
-                DisplayHandler.Order.DOWN -> i += it.getText().getHeight()
                 DisplayHandler.Order.UP -> i -= it.getText().getHeight()
+                DisplayHandler.Order.DOWN -> i += it.getText().getHeight()
             }
         }
 
         height = i
+
+        onPostDraw?.trigger(arrayOf(this))
+
     }
 
     internal abstract fun createDisplayLine(text: String): DisplayLine
 
+    internal abstract fun getLoader(): ILoader
```

Comment:
Inside `Impl.kt`, add `override fun getLoader` for JSDisplay like the other classes in there.